File: preprocessing/CSV_TSV_to_RDS.py
import numpy as np
import pandas as pd
import pyreadr
import argparse

### Initialize the parsed arguments
parser = argparse.ArgumentParser(description='Convert CSV or TSV file data frame to RDS file')
parser.add_argument('--type', action='store',help='only TSV or CSV')
parser.add_argument('--full_file_path', action='store',help='The CSV or TSV input file name (with the extention .csv or .tsv) file and its full path to convert into RDS')
parser.add_argument('--output_full_file_path', action='store',help='The RDS (with the extention .rds) and its full path to convert from CSV/TSV')
args = parser.parse_args()
full_file_path= args.full_file_path
output_full_file_path= args.output_full_file_path

### Load the data
if type=='CSV':
    df = pd.read_csv(full_file_path,index_col = 0 )
else:
   df = pd.read_csv(full_file_path,sep='\t',index_col = 0)

df = df.reset_index()
# Rename the new column if needed (default will be 'index')
df.rename(columns={'index': 'rownames'}, inplace=True)

### Save the file
pyreadr.write_rds(output_full_file_path, df)

File: preprocessing/CSV_TSV_to_RDS.R
library(tidyverse)
inputFile <- 'preprocessed_data/l1000_all_genes_lvl3_drugs_with_targets_exemplar.tsv'
outputFile <- 'preprocessed_data/l1000_all_genes_lvl3_drugs_with_targets_exemplar.rds'
type <- 'TSV'

### Load to file
if (type=='TSV'){
  df <- data.table::fread(inputFile,sep = "\t") %>% column_to_rownames('V1')
}else{
  df <- data.table::fread(inputFile) %>% column_to_rownames('V1')
}

### Load file
saveRDS(df,outputFile)